### ISSUE TITLE
Hardware 

### ISSUE THREAD

**Issue by swapdha** [Hypha-ROS/hypha-racecar#4]
Here are my 2 question 
(1)I have" HSP 94186" rc car , it is "1/16" scale , hardware similar to "HSP 94123" but it's small scale. I want to know can  i use "HSP 94186" instead of buying "HSP 94123" . I know it is smaller but i can manage with hardware intergartion.  
(2)Have you change "HSP 94123" stock -esc with "TAMIYA ESC TBLE-02S" ,steering servo with 
"FUTABA S9151 digital servo" and battery with "7.4v 5000 Mah battery"or you bought a empty 1/10 car frame and added all this parts mentioned in build doc pdf-"https://github.com/Hypha-ROS/hypha-racecar/blob/ver.1-release/document/Hypha-ROS-RaceCar-Tutorial_V1.1.pdf"

**Comment by haochihlin** [comment]
Hi @swapdha 
For your questions, please find detail as below:
1. Definitely yes, actually, we are moving all source codes to our new 1/20 scaled minicar platform.
The thing you need to take care is that you have modify car related parameters in launch file
(e.g. the length of car for L1 controller, etc)

2. Yes, since the performance of ESC sent with HSP 94123 is pretty bad, we decided to use TAMIYA ESC.
But we also tested with hobbywing brushless ESC in our RaceCar ver2. For the servo, you can replace it with any digital based one.

Please feel free to comment here if you have any question.
HaoChih

**Comment by swapdha** [comment]
Is it ok to use all stock parts(esc,sterring servo,battery) of HSP 94186  or I need to change.

**Comment by subodh-malgonde** [comment]
@jim1993 Looking forward to the setup guide for 1/20 scale car. Can you give us an idea when you will release it?

I have a question about the weight of electronics - will the chassis and suspension of a 1/20 scale car be able to carry the same SBC, lidar, battery and camera as that for a 1/10 scale car? Can you please share the bill of materials for the 1/20 scale car?

**Comment by haochihlin** [comment]
Hi @subodh-malgonde 
About the 1/20 scale car (we called HyphaROS minicar), well, currently we are working on porting custom MPC (model predictive controller) node to odroid xu4 environment. So if everything is on schedule, we could release our first beta version by the end of August (I hope so :D ).

For the hardware architecture, we still use odroid xu4 as SBC, plus YDLidar X4 for localization and STM32 MCU for motor speed and servo control (our minicar is ackermann design with encodered rear-wheel drives).

**Comment by subodh-malgonde** [comment]
Thanks for getting back @jim1993. I have a couple more questions, if you don't mind.

1. If I may ask, which 1/20 model are your using? 

2. Why is Ackermann design important? I know that full sized cars have Ackermann design. However why is that important for a smaller scale version which is likely to be used indoors?

**Comment by haochihlin** [comment]
Hi
I'm glad you asked:
1. We took a low-cost ackermann scaled model car from Taobao, then add lidar and odroid on it:
![20180629_220609](https://user-images.githubusercontent.com/8316182/42096801-f01c5f72-7be8-11e8-8f3a-dc4578d3ed07.jpg)
![20180629_220619](https://user-images.githubusercontent.com/8316182/42096802-f04ac0b0-7be8-11e8-8352-31b730f02483.jpg)

2. Good question :D 
As you mentioned, for small platform, differential drive is quite enough, so it is one of reasons we created the "HyphaROS-MiniBot", which is exactly a different drive ROS based robot.
However, my background is "Control & Dynamic", therefore, I want to implement what I have learned from theories on real platform. For example: pure pursuit, model predictive control, LQR, etc. To illustrate these advanced control laws on this kind of low-cost car could be a great as well as affordable way for students to learn and experience what are the physical meaning of those algorithms taught by Profs or lecturers. 
In short: just for fun ~~~!!!
